### PR TITLE
Minor refactoring and increase of coverage (Validator tests)

### PR DESCRIPTION
**Pull request by satoshi101**
Minor refactoring and increase of coverage. It adds Validator tests.

### CODE REVIEW

File: tests/Data/Test_Transaction.cpp [hunk 1]
```diff
@@ -43,21 +43,15 @@ BOOST_AUTO_TEST_CASE(test1)
 
     Address toAddr;
 
-    Mediator* m = nullptr;
-    unique_ptr<ValidatorBase> m_validator = make_unique<Validator>(*m);
+    NoopMediator nm;
```
Comment (lookfwd) [inline tests/Data/Test_Transaction.cpp]: Move `NoopMediator` in `Test_Transaction.cpp`

File: src/libValidator/Validator.cpp [hunk 2]
```diff
@@ -17,17 +17,70 @@
...
+unsigned int MediatorAdapter::getNumShards() const
+{
+    return m_mediator.m_node->getNumShards();
+}
+const char* MediatorAdapter::currentEpochNumAsString() const
+{
+    return to_string(m_mediator.m_currentEpochNum).c_str();
+}
+
+NoopMediator::~NoopMediator() {}
```
Comment (digiront) [inline src/libValidator/Validator.cpp]: Move all `NoopMediator` code to test (if it's not used in actual code)

File: src/libValidator/Validator.cpp [hunk 3]
```diff
@@ -17,17 +17,70 @@
...
+    return m_mediator.m_node->getShardID();
+}
+
+unsigned int MediatorAdapter::getNumShards() const
+{
+    return m_mediator.m_node->getNumShards();
+}
+const char* MediatorAdapter::currentEpochNumAsString() const
+{
+    return to_string(m_mediator.m_currentEpochNum).c_str();
```
Comment (Gnnng) [inline src/libValidator/Validator.cpp]: Seems a freed pointer is being returned

File: tests/Validator/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,61 @@
+if(CMAKE_CONFIGURATION_TYPES)
```
Comment (Gnnng) [inline tests/Validator/CMakeLists.txt]: I'm afraid this `CMakeLists.txt` needs some changes. I could see that it's probably developed from [tests/POW/CMakeLists.txt](https://github.com/Zilliqa/Zilliqa/blob/de2e47879d620aade220176d0ce9349cc8b7bdf5/tests/POW/CMakeLists.txt). Although it works, it's actually a bit messy and a better CMakeLists.txt example is [tests/Data/CMakeLists.txt](https://github.com/Zilliqa/Zilliqa/blob/master/tests/Data/CMakeLists.txt).

File: src/libValidator/Validator.cpp (hunk 3 above)
Comment (Gnnng) [reply src/libValidator/Validator.cpp]: returning a `std::string` would be better

**Comment by Gnnng** [comment]
@satoshi101 Can you address these issues before we can merge this PR?
- resolve`MediatorAdapter::currentEpochNumAsString` use-after-free issue
- clean up tests/Validator/CMakeLists.txt 
- move `NoopMediator` to tests

**Comment by Gnnng** [comment]
These issues are fixed at https://github.com/Zilliqa/Zilliqa/commits/fix/324. The conflict with master takes some time to fix.

Waiting for PR author to adopt those changes

File: src/libValidator/Validator.cpp (hunk 2 above)
Comment (satoshi101) [reply src/libValidator/Validator.cpp]: Great idea

File: src/libValidator/Validator.cpp (hunk 3 above)
Comment (satoshi101) [reply src/libValidator/Validator.cpp]: Great catch! Thanks!

**Comment by satoshi101** [comment]
Done merging latest master

**Pull request closed without merging**